Adding python 3.11 to build_tests action

kbonney opened the pull request:
Adding 3.11 to GH action, as discussed.

kbonney committed:
Adding python 3.11 to build_tests action

kbonney committed:
fix typo

kbonney committed:
Adding installation of req.txt in order to properly make docs

kaklise commented on documentation/developers.rst:
```diff
@@ -57,7 +57,13 @@ New content is marked `Draft`.
+repository::
+
+	pip install -r requirements.txt
+
```
Seems like we might want to add this information to the installation page, since users will want the optional dependencies for functionality other than building docs.

kbonney replied:
Agreed. Do we want to provide a way to selectively install the optional packages versus what is needed to run pytest versus what is needed to build the documentation?

kbonney committed:
Specifying language to address docs warning

kbonney committed:
Fixing link to valve layer and link to ipynb

kbonney committed:
pytest instructions given in developers.rst. recommending to remove from installation section.

kbonney committed:
Fixing warnings raised by duplicate ref labels. They are assigned once in the original .rst files and then again when included with include::

kbonney committed:
Addressing doc warning

kbonney committed:
small typo

kbonney committed:
Trying to fix API doc formatting. Intersect docstring was rendering poorly

kbonney committed:
Adding newlines to fix api doc formatting

kbonney commented:
- Fixed a couple typos and some formatting issues in docstrings that were causing some api docs to render poorly. Fixed some links on the criticality page.
- I also added the latest whatsnew (060) to be included in the release notes page
- Noticed that pytest instructions are given in developers.rst and installation.rst. Recommending we only have it in developers.rst.

kbonney committed:
Removing numpydoc since no longer used

kbonney committed:
Adding updates to req.txt

kbonney committed:
Reverting docs build instructions to previous text. req.txt installed as part of the developers installation process. Adding pytest wntr back to developer instructions. Mentioning the installation of req.txt to install optional dependencies.

kbonney committed:
This was meant to be a part of the last commit.

kbonney committed:
Changing references to use class instead of func or mod.

kbonney committed:
fixing references to functions
